src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -244,6 +244,18 @@ public function finishView(FormView $view, FormInterface $form, array $options)
                 $childView->vars['full_name'] = $childName;
             }
         }
+
+        // We have to override parent {@link FormType} condition of the empty_view option here,
+        // even if there is no choices the field view may not be empty because:
+        //
+        //  - a select input is not "compound" but should be considered as any empty HTML tag,
+        //
+        //  - when expanded it could hold a placeholder which has been counted as a children,
+        //    we should ignore it in any case since its role is to be a "null" alternative to
+        //    choices, not a choice itself. Think of "Choose something" with no choices.
+        if (array() === $view->vars['choices'] && array() === $view->vars['preferred_choices']) {
```

Comment:
Why is this check needed? Can't we decide in the view whether to output the text?